tests/System/support/commands/db.js

```diff
@@ -439,21 +439,41 @@ Cypress.Commands.add('db_createMenuItem', (menuItemData) => {
     language: '*',
     params: '',
     img: '',
+    lft: 0,
+    rgt: 0,
   };
 
-  // Create the data to insert
-  const menuItem = { ...defaultMenuItemOptions, ...menuItemData };
-
-  // Extract the component from the link
-  const component = (new URLSearchParams(menuItem.link.replace('index.php', ''))).get('option');
-
-  // Search for the component
-  return cy.task('queryDB', `SELECT extension_id FROM #__extensions WHERE name = '${component}'`).then((id) => {
-    // Get the correct component id from the extensions record
-    menuItem.component_id = id[0].extension_id;
+  // Create space for rgt and lft
+  cy.task('queryDB', 'SELECT rgt FROM #__menu WHERE id = 1').then((myrgt) => {
+    defaultMenuItemOptions.lft = myrgt[0].rgt;
+    defaultMenuItemOptions.rgt = myrgt[0].rgt + 1;
+
+    const menuItem = { ...defaultMenuItemOptions, ...menuItemData };
+    // Extract the component from the link
+    const component = (new URLSearchParams(menuItem.link.replace('index.php', ''))).get('option');
+    return cy.task('queryDB', `SELECT extension_id FROM #__extensions WHERE name = '${component}'`).then((id) => {
+      // Get the correct component id from the extensions record
+      menuItem.component_id = id[0].extension_id;
+      return cy.task('queryDB', `UPDATE #__menu SET rgt = rgt + 2 WHERE rgt >= '${defaultMenuItemOptions.lft}'`)
+        .then(() => cy.task('queryDB', `UPDATE #__menu SET lft = lft + 2 WHERE lft > '${defaultMenuItemOptions.rgt}'`))
+        .then(() => cy.task('queryDB', createInsertQuery('menu', menuItem)).then(async (info) => info.insertId));
```

Comment:
```suggestion
        .then(() => cy.task('queryDB', createInsertQuery('menu', menuItem)))
        .then(async (info) => info.insertId);
```

Should this not be like that? Otherwise you don't return the insertId.

Reply:
`insertId` is returned
otherwise  the other test should fail
https://github.com/joomla/joomla-cms/blob/4d6856ca5b370a93fb970775a9929d3cfac5f261/tests/System/integration/api/com_menus/SiteMenuItems.cy.js#L18

Reply:
You are right, but I would still take it out as it is easier then to read.